src/DynamoCoreWpf/ViewModels/Search/SearchViewModel.cs

```diff
@@ -843,6 +855,56 @@ private static string MakeShortCategoryString(string fullCategoryName)
 
         #endregion
 
+        #region Key navigation
+
+        public enum MovementDirection
+        {
+            Forward, Back
+        }
+
+        /// <summary>
+        /// Executes selected item in search UI.
+        /// </summary>
+        public void ExecuteSelectedItem()
+        {
+            var selected = FilteredResults.FirstOrDefault(item => item.IsSelected);
+
+            if (selected != null)
+            {
+                selected.ClickedCommand.Execute(null);
+            }
+        }
+
+        /// <summary>
+        /// When down key is pressed, selected element should move forward.
+        /// When up key is pressed, selected element should move back.
+        /// </summary>
+        public void MoveSelection(MovementDirection direction)
+        {
+            var oldItem = FilteredResults.FirstOrDefault(item => item.IsSelected);
+            if (oldItem == null) return;
+
+            int newItemIndex = FilteredResults.IndexOf(oldItem);
+
+            if (direction == MovementDirection.Forward)
+            {
+                newItemIndex++;
+            }
+            else
+            {
+                newItemIndex--;
+            }
+
+            if (newItemIndex < 0 || newItemIndex >= FilteredResults.Count()) return;
```

Comment:
True.Should it be newItemIndex <= 0 in the first if statement? by defualt the first item is selected. So if the first item is reached then just ignore the up direction. You can control that in the direction instead of multiple if statments.
If the direction is down, then check whether the newitemindex >=0, for up check newitemindex <= count().